docs/performance.md

```diff
@@ -0,0 +1,254 @@
+# Tessera Storage Performance
+
+This document describes the performance of each Trillian Tessera storage implementation under different conditions.
+
+## MySQL
+
+### GCP Free Tier VM Instance
+
+**e2-micro**
+
+- vCPU: 0.25-2 vCPU (1 shared core)
+- Memory: 1 GB
+- OS: Debian GNU/Linux 12 (bookworm)
+
+#### Result
+
+```
+┌──────────────────────────────────────────────────────────────────────┐
+│Read (8 workers): Current max: 0/s. Oversupply in last second: 0      │
+│Write (256 workers): Current max: 409/s. Oversupply in last second: 0 │
+│TreeSize: 240921 (Δ 307qps over 30s)                                  │
+│Time-in-queue: 86ms/566ms/2172ms (min/avg/max)                        │
+│Observed-time-to-integrate: 516ms/1056ms/2531ms (min/avg/max)         │
+└──────────────────────────────────────────────────────────────────────┘
+```
+
+The bottleneck is at the dockerized MySQL instance, which consumes around 50% of the memory.
```

Comment:
Could you connect the dots for me as to why 50% memory usage is the bottleneck? I suspect you know something I don't :)

Reply:
If you scroll down a bit, you will see `mysqld` consumed 49.7% of the memory with 74.5 MiB free in total. The CPU usage is low. The swap memory is low as well, so the machine can barely work...

Reply:
That's why I did another round of test with the same VM instance and Cloud SQL. The numbers improved a lot.

Reply:
Could the bottleneck be very restricted IO limiting MySQL's performance?

Reply:
It could be as IO increases when memory is running low. The default MySQL docker image is used without any performance tuning.